### PR TITLE
add cdk for Local Office Search API

### PR DESCRIPTION
**Pull request by cnorthwood**

### CODE REVIEW

File: cdk/app/local_office_search_api.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            labels={"app.kubernetes.io/name": "local-office-search-api"},
+        )
+        lss_data_bucket.grant_read(self._service_account)
+        geo_data_bucket.grant_read(self._service_account)
+
+        self._cdk8s_app = App()
+
+        rds_secret_source = ExternalSecretSource(
+            source_secret=db_credentials.secret_name,
+            secret_mappings={"password": "DB_PASSWORD"},
```
Comment (michelesr) [inline cdk/app/local_office_search_api.py:88]: why only the password, the app would need the endpoints and the username too, right?

Comment (cnorthwood) [reply cdk/app/local_office_search_api.py:88]: I've not considered those to be secrets so I'm passing them directly through from the DB infra stack and then using `from_value` https://github.com/citizensadvice/local-office-search-api/blob/34872647b8e977dd4eb2f11cc1676a166d7a6b8d/cdk/app/chart.py#L167-L170

Comment (michelesr) [reply cdk/app/local_office_search_api.py:88]: but then you're not reading the credentials from the k8s secret, including the password, you're injecting them directly, it doesn't seem right, you should be using `envFrom` to get it from the secret... I would put all the db credentials together to make it simpler but that's up to you

Comment (michelesr) [reply cdk/app/local_office_search_api.py:88]: or maybe cdk8s does that for you? I'm not familiar with the syntax

Comment (michelesr) [reply cdk/app/local_office_search_api.py:88]: ah it's a cdk8s+ thing ...

**Review by michelesr — APPROVED** [review]

**Pull request merged**